cmd/directpv/node-controller.go

```diff
@@ -17,39 +17,61 @@
 package main
 
 import (
+	"context"
 	"errors"
 	"os"
 
-	"github.com/minio/directpv/pkg/admin"
 	"github.com/minio/directpv/pkg/consts"
+	"github.com/minio/directpv/pkg/initrequest"
+	"github.com/minio/directpv/pkg/node"
 	"github.com/minio/directpv/pkg/sys"
 	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
 )
 
-var nodeAPIPort = consts.NodeAPIPort
-
-var nodeAPIServerCmd = &cobra.Command{
-	Use:           consts.NodeAPIServerName,
-	Short:         "Start node API server.",
+var nodeControllerCmd = &cobra.Command{
+	Use:           consts.NodeControllerName,
```

Comment:
What is the benefit of having separate container for this?

Reply:
isolating the concerns and reducing the failure domain. For example, node discovery error shouldn't fail the csi operations. The csi containers should be running fine.